src/plugins/intel_gpu/tests/unit/test_cases/propagate_constants_gpu_test.cpp

```diff
@@ -57,3 +60,49 @@ TEST(propagate_constants, copy_dependecies_from_nodes) {
 TEST(propagate_constants, copy_dependecies_from_nodes_cached) {
     test_copy_dependecies_from_nodes<ov::float16>(true);
 }
+
+TEST(propagate_constants, permute_1_0_reorder_fc) {
+    auto& engine = get_test_engine();
+
+    auto input2_layout_dyn = layout{ ov::PartialShape{ -1, 32 }, data_types::f16, format::bfyx };
+
+    auto input = engine.allocate_memory({ { 2, 32 }, data_types::f16, format::bfyx });
+    auto input2 = engine.allocate_memory({ { 2, 32 }, data_types::f16, format::bfyx });
+    auto weights = engine.allocate_memory({{ 32, 2 }, data_types::f32, format::bfyx });
+
+    tests::random_generator rg(GET_SUITE_NAME);
+    auto input_data = rg.generate_random_2d<ov::float16>(2, 32, -1, 1);
+    auto input2_data = rg.generate_random_2d<ov::float16>(2, 32, -1, -1);
+    auto weights_data = rg.generate_random_2d<float>(32, 2, -1, 1);
+
+    set_values(input, flatten_2d(format::bfyx, input_data));
+    set_values(input2, input2_data);
+    set_values(weights, flatten_2d(format::bfyx, weights_data));
+
+    topology topology(
+        input_layout("input", input->get_layout()),
+        input_layout("input2", input2_layout_dyn),
+        data("weights", weights),
+        permute("permute_test", input_info("weights"), {1, 0}),
+        reorder("reorder_dt", input_info("permute_test"), format::any, data_types::f16, std::vector<float>()),
+        fully_connected("fc1", input_info("input"), { "reorder_dt" }, "", data_types::f16),
+        fully_connected("fc2", input_info("input2"), { "reorder_dt" }, "", data_types::f16)
+    );
+
+    ExecutionConfig config = get_test_default_config(engine);
+    config.set_property(ov::intel_gpu::optimize_data(true));
+    config.set_property(ov::intel_gpu::allow_new_shape_infer(true));
+
+    if (engine.get_device_info().supports_immad) {
+        ov::intel_gpu::ImplementationDesc fc_impl = { format::bfyx, "", impl_types::onednn };
+        config.set_property(ov::intel_gpu::force_implementations(ov::intel_gpu::ImplForcingMap{ {"fc1", fc_impl} }));
+    }
+
+    cldnn::network network(engine, topology, config);
+    network.set_input_data("input", input);
+    network.set_input_data("input2", input2);
+
+    auto outputs = network.execute();
+    auto output = outputs.at("fc1").get_memory();
+    cldnn::mem_lock<ov::float16> output_ptr(output, get_test_stream());
```

Comment:
It is just declaring the output pointer.
Shouldn't we test something more with this?;

Reply:
Done